src/auth/googleauth.ts

```diff
@@ -672,6 +667,10 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
   ): JSONClient {
     let client: JSONClient;
 
+    // user's preferred universe domain
+    const preferredUniverseDomain =
```

Comment:
Not a huge fan of this pattern, however the existing `.fromJSON` pattern makes it difficult to add preferred options. We should consider a refactor to `[AuthClient].fromJSON(json, overrrides)` later